baml_language/scripts/smoke_cpp_tarball.sh

```
@@ -0,0 +1,67 @@
+
+# Resolve the tarball argument relative to the caller's directory before
+# changing into the workspace.
+TARBALL="${1:-}"
+if [ -n "$TARBALL" ]; then
+    TARBALL="$(cd "$(dirname "$TARBALL")" && pwd)/$(basename "$TARBALL")"
+fi
+
+cd "$(dirname "$0")/.."
+
+if [ -z "$TARBALL" ]; then
+    TARBALL="$(ls -t target/cpp-dist/baml-cpp-*.tar.gz 2> /dev/null | head -1)"
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)